index.bs

```diff
@@ -9024,6 +9070,419 @@ worker=] algorithm:
+    Extensible,
+}
+</pre>
+
+The <code>storage.PartitionKey</code> type represents a [=storage partition key=].
+
+The following <dfn>table of standard storage partition key attributes</dfn>
+enumerates attributes with well-known meanings which a [=remote end=] may
+choose to support. An implementation may define additional [=extension storage
+partition key attributes=].
+
+<table>
+  <tr>
+    <th>Attribute
+    <th>Definition
+  <tr>
+    <td>"<code>userContext</code>"
+    <td>
+      A user context id
+
+      Issue(w3c/webdriver-bidi#570): User contexts are in the process of being
+      specified.
+  <tr>
+    <td>"<code>sourceOrigin"</code>
+    <td>The [=serialization of an origin=] from which the storage partition is to be accessed
+</table>
+
+[=Remote ends=] may support any number of <dfn>extension storage partition key
+attributes</dfn>. In order to avoid conflicts with other implementations, these
+attributes must begin with a unique identifier for the vendor and user-agent
+followed by U+003A (:).
+
+A [=remote end=] has a [=/map=] of <dfn>default values for storage partition
+key attributes</dfn> which contains zero or more implementation-defined
+entries. Each key must be a member of the [=table of standard storage partition
+key attributes=] or an [=extension storage partition key attribute=].
+
+Note: Default values are used by [=storage.getCookies=], [=storage.setCookie=],
+and [=storage.deleteCookies=] when the [=local end=] does not specify a value.
+
+A [=remote end=] has a [=/list=] of <dfn>required partition key
+attributes</dfn> which contains zero or more implementation-defined entries.
+Each key must be a member of the [=table of standard storage partition key
+attributes=] or an [=extension storage partition key attribute=].
+
+<div algorithm>
+To <dfn>expand a storage partition spec</dfn> given |partition spec|:
+
+1. If |partition spec| is null:
+
+  1. Let |partition key| be an empty [=/map=].
+
+1. Otherwise, if |partition spec|["<code>type</code>"] is "<code>context</code>":
+
+  1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=] given |partition spec|.
+
+  1. Let |partition key| be the [=storage partition key|key=] of |browsing context|'s [=associated storage partition=].
+
+  1. Return [=success=] with data |partition key|.
+
+1. Otherwise:
+
+  1. Assert: |partition spec|["<code>type</code>"] is "<code>storageKey</code>".
+
+  1. Let |partition key| be |partition spec|.
+
+1. For each |name| → |default value| in the [=default values for storage partition key attributes=]:
+
+  1. If |partition key|[|name|] does not [=map/exist=]:
+
+    1. [=map/Set=] |partition key|[|name|] to |default value|.
+
+1. For each |name| in the remote end's [=required partition key attributes=]:
+
+  1. If |partition key|[|name|] does not [=map/exist=]:
+
+    1. Return [=error=] with [=error code=] [=underspecified storage partition=].
+
+1. Return [=success=] with data |partition key|.
```

Comment:
This will still have a `type` attribute. It will also still include anything that was supplied but isn't used. We really want to end up with just the things that actually form part of the used keys.

Reply:
Addressed via commit 35d7fad.